errors.go

```diff
@@ -317,3 +319,72 @@ func (e *RFC6749Error) WithDescription(description string) *RFC6749Error {
 	err.Description = description
 	return &err
 }
+
+func (e *RFC6749Error) Sanitize() *RFC6749Error {
+	err := *e
+	err.Debug = ""
+	return &err
+}
+
+// GetDescription returns a more description description, combined with hint and debug (when available).
+func (e *RFC6749Error) GetDescription() string {
+	description := e.Description
+	if e.Hint != "" {
+		description += "\n\n" + e.Hint
+	}
+	if e.Debug != "" {
+		description += "\n\n" + e.Debug
+	}
+	return description
+}
+
+// RFC6749ErrorJson is a helper struct for JSON encoding/decoding of RFC6749Error.
+type RFC6749ErrorJson struct {
+	Name        string `json:"error"`
+	Verbose     string `json:"error_verbose"`
```

Comment:
Feel free to suggest any other name. I wanted to to keep the original description around so that we can define `UnmarshalJSON`.

Reply:
I would probably expect verbose to be more verbose than the other description. Given that we‘re changing the API in a breaking manner and that all OAuth2 clients who rely on this original prameter would need to be updated anyways I think we can simply omit it :)

Reply:
This is something else. We are breaking `Description`, but this field is here so that one can marshal and unmarshal the error object. Without this field it becomes tricky (I would have to unparse the multi-line description, not too hard, but slightly fragile).

Maybe then `ShortDescription`?

I would be able to break everything, I would go for:

* Code
* Name
* Description
* Hint
* StatusCode
* Debug

Reply:
I see, in that case the semantics make sense!